### PR TITLE
feat(complex): add QR decomposition support

### PR DESCRIPTION
**Pull request by polvalente**
Related to #692

### CODE REVIEW

File: nx/test/nx/lin_alg_test.exs [hunk 1]
```diff
@@ -159,10 +159,36 @@ defmodule Nx.LinAlgTest do
       assert_all_close(Nx.dot(q, r), t, atol: 1.0e-10)
     end
 
+    test "works with complex matrix" do
+      import Nx, only: :sigils
```
Comment (polvalente) [inline nx/test/nx/lin_alg_test.exs:163]: is it ok if I import this at the top of the module?

File: nx/test/nx/lin_alg_test.exs [hunk 2]
```diff
@@ -159,10 +159,36 @@ defmodule Nx.LinAlgTest do
...
+        0 2 -1i
+        1 1 1
+      ]
+
+      {q, r} = Nx.LinAlg.qr(t)
+
+      assert_all_close(q, ~M[
+        -0.7070 0.2357  0.6666
+         0      -0.9428   0.3333
+        -0.7070 -0.2357  -0.6666
```
Comment (polvalente) [inline nx/test/nx/lin_alg_test.exs:176]: During tests, I needed to copy and paste complex tensors from the terminal but couldn't just paste them into the sigil due to the brackets for each row.

Perhaps we can allow for that through a regex? Something like:

String.replace(string, ~r/\[([^\]+)\]/, "\\1")

File: nx/test/nx/lin_alg_test.exs (hunk 1 above)
Comment (josevalim) [reply nx/test/nx/lin_alg_test.exs:163]: Totally!

File: nx/test/nx/lin_alg_test.exs (hunk 2 above)
Comment (josevalim) [reply nx/test/nx/lin_alg_test.exs:176]: Another option is to print the tensor differently. Using a similar format as ~M. Higher dimensional tensors are printed as "lists" of matrices.

**Review by josevalim — APPROVED** [review]

File: nx/test/nx/lin_alg_test.exs (hunk 2 above)
Comment (polvalente) [reply nx/test/nx/lin_alg_test.exs:176]: So we would print all tensors like this every time?

Comment (josevalim) [reply nx/test/nx/lin_alg_test.exs:176]: I don't know, somethjing to discuss :)

**Pull request merged**